lib/components/drawer/right_drawer.dart

```diff
@@ -0,0 +1,206 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:rtchat/components/channel_search_dialog.dart';
+import 'package:rtchat/components/drawer/quicklinks_listview.dart';
+import 'package:rtchat/models/audio.dart';
+import 'package:rtchat/models/channels.dart';
+import 'package:rtchat/models/layout.dart';
+import 'package:rtchat/models/tts.dart';
+import 'package:rtchat/models/user.dart';
+
+class _ChannelPickerValue {
+  final Channel? channel;
+  final bool isAdd;
+
+  const _ChannelPickerValue({this.channel, this.isAdd = false});
+}
+
+class RightDrawer extends StatelessWidget {
+  const RightDrawer({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Drawer(
+      child: ListView(
+        padding: EdgeInsets.zero,
+        children: [
+          BackdropFilter(
+            filter: ImageFilter.blur(sigmaX: 5, sigmaY: 5),
+            child: Container(
+              color: Colors.black.withOpacity(0),
+            ),
+          ),
+          SizedBox(
+            height: 128,
+            child: DrawerHeader(
+              padding: EdgeInsets.zero,
+              child: Padding(
+                padding: const EdgeInsets.only(
+                  left: 20,
+                  top: 8,
+                  bottom: 8,
+                ),
+                child: _buildChannelListView(),
+              ),
+            ),
+          ),
+
+          // quicklinks
+          QuicklinksListView(),
+
+          const Divider(
+            indent: 30,
+            endIndent: 30,
+            thickness: 2,
+          ),
+
+          // setting
+          Consumer<LayoutModel>(builder: (context, layoutModel, child) {
+            if (layoutModel.locked) {
```

Comment:
I believe this one is flipped.
```suggestion
            if (!layoutModel.locked) {
```